### PR TITLE
Add maven wrapper to scala project.

### PR DESCRIPTION
**Pull request by frankfliu**
Add maven wrapper to scala project, this removed maven dependency from host machine.
And this will also eliminate build script compatibility issue with different maven version.

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [X] Changes are complete (i.e. I finished coding on this PR)
- [X] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [X] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [X] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### CODE REVIEW

**Comment by lebeg** [comment]
> As per previous discussion...

@marcoabreu Please provide the link. 

I would consider reading answers in [LEGAL-427](https://issues.apache.org/jira/browse/LEGAL-427) carefully as well.

**Comment by Roshrini** [comment]
@nswamy Can you please take a look?
@mxnet-label-bot Add [pr-awaiting-review, Scala]

**Comment by frankfliu** [comment]
@marcoabreu 
checkin maven wrapper or gradle wrapper is very common practice in most of modern java projects.
check in binary files are bad in general, which is not able diff while updating, and will increase git repo size rapidly. But It should be OK for tools since they are not part of source code and won't be updated frequently. And maven wrapper and gradle wrapper jar files are tiny.

You can find many apache project with maven wrapper checked in github:
Here is a link to apache log4j project: https://github.com/apache/logging-log4j2/tree/master/.mvn/wrapper

**Comment by anirudhacharya** [comment]
@marcoabreu ping for a relook at this PR.

File: scala-package/.mvn/wrapper/MavenWrapperDownloader.java [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+            System.out.println("Done");
+            System.exit(0);
+        } catch (Throwable e) {
+            System.out.println("- Error downloading");
+            e.printStackTrace();
+            System.exit(1);
+        }
+    }
+
+    private static void downloadFileFromURL(String urlString, File destination) throws Exception {
```
Comment (lanking520) [inline scala-package/.mvn/wrapper/MavenWrapperDownloader.java]: Please add some retries in here to reduce the failure

Comment (piyushghai) [reply scala-package/.mvn/wrapper/MavenWrapperDownloader.java]: I don't think it is needed. It's code coming in from Maven Wrapper package.

**Comment by lanking520** [comment]
@frankfliu Please raise a discussion on dev@ to discuss about uploading the binary

**Comment by marcoabreu** [comment]
Thanks @lanking520 , that's a great idea. We have been advised by out mentors in the past to specifically remove the maven wrapper binary, but it might be good to bring it up again.

**Comment by mjwall** [comment]
Marco,

Agree you shouldn't check in binaries, but you can exclude the maven
wrapper binary from source control and it will be downloaded when needed.
See https://github.com/takari/maven-wrapper#usage-without-binary-jar.
Interested to read any links referencing those discussions if you have them.

Mike

**Comment by marcoabreu** [comment]
Ah, that's an excellent alternative then :)

There you go: https://lists.apache.org/thread.html/d409819b554745937103b60d36ad3f5b2b1300fe8ae826fb0e665f7f@%3Cdev.mxnet.apache.org%3E

**Comment by marcoabreu** [comment]
Original source: https://lists.apache.org/thread.html/2bf07a377cac35d4940f0fbf7f647f46a38d2845e5dbb3c650f367d7@%3Cgeneral.incubator.apache.org%3E

**Comment by mjwall** [comment]
Great, thanks for the links.  I don't see the issue if the jar file is not
included in the source and not included in the distirbution, but interested
to hear other's interpretations.  LEGAL-288 to me is about including that
jar file.  Was this asked on the dev list?

**Comment by marcoabreu** [comment]
We had it in the past and were asked to remove it.

If there is an alternative to a binary file - which is the case according to your link - I'm also in favour to not have them checked in at all as otherwise users will have to manually download them and we have to manually remove them.

**Comment by vandanavk** [comment]
@mxnet-label-bot update [Scala, pr-awaiting-merge]

**Comment by lanking520** [comment]
@frankfliu please add the remaining items in JIRA as a backlog item

**Pull request merged**